src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,214 @@
+
+            return true;
+        } else if (userInput.split(" ")[0].equals("mark")) {
+            String[] input = userInput.split(" ");
+            Task task =  Task.getTaskList()[Integer.parseInt(input[1]) - 1];
+            task.markDone();
+
+            String output = Ui.createLine()
+                    + "       I have marked the following task as done! :D \n"
+                    + "       " + task + "\n"
+                    + Ui.createLine();
+
+            System.out.println(output);
+
+            return true;
+        } else if (userInput.split(" ")[0].equals("unmark")) {
+            String[] input = userInput.split(" ");
+            Task task =  Task.getTaskList()[Integer.parseInt(input[1]) - 1];
+            task.markNotDone();
+
+            String output = Ui.createLine()
+                    + "       OK I have marked the following task as not done yet! :D \n"
+                    + "       " + task + "\n"
+                    + Ui.createLine();
+            System.out.println(output);
+
+            return true;
+        } else if (userInput.split(" ")[0].equals("delete")) {
+            String[] input = userInput.split(" ");
+            Task task = Task.getTaskList()[Integer.parseInt(input[1]) - 1];
+            task.deleteTask(task);
+
+            String output = Ui.createLine()
+                    + "       OK I have delete the following task! :D \n"
+                    + "       " + task + "\n"
+                    + "       " + "Now you have " + Task.getCounter() + " tasks in your list.\n"
+                    + Ui.createLine();
+            System.out.println(output);
+
+            return true;
+        } else if (userInput.split(" ")[0].equals("find")) {
+            String[] input = userInput.split(" ");
+            Integer[] foundTask = Task.findTask(userInput.substring(5));
+
+            if(input.length > 1) {
+
+                String output = Ui.createLine()
+                        + "       OK I have found the following task! :D \n"
+                        + Task.printArray(foundTask) + "\n"
+                        + Ui.createLine();
+                System.out.println(output);
+
+            } else {
+                String errorMsg = Ui.createLine()
+                        + "       You have entered the find command.\n"
+                        + "       What do you want to find?\n"
+                        + Ui.createLine();
+                System.out.println(errorMsg);
+            }
+
+
+            return true;
+
+        } else {
+
+            try {
+                String newTaskMessage = Parser.createTask(userInput);
+                System.out.println(newTaskMessage);
+            } catch (InvalidTypeException e) {
+                String errorMsg = Ui.createLine()
+                        + "       Opps, the command \"" + userInput + "\" is not supported :(\n"
+                        + Ui.createLine();
+                System.out.println(errorMsg);
```

Comment:
Would there be a way to use the error messages stored inside the exceptions instead of creating a new error message here? Otherwise, is there a need for the exceptions to have a message in their constructors?